core-plugins/src/main/java/co/cask/hydrator/plugin/batch/joiner/JoinerConfig.java

```diff
@@ -89,4 +104,79 @@ public String getJoinKeys() {
   public String getRequiredInputs() {
     return requiredInputs;
   }
+
+
+  Map<String, List<String>> getPerStageJoinKeys() {
+    Map<String, List<String>> stageToKey = new HashMap<>();
+
+    if (Strings.isNullOrEmpty(joinKeys)) {
+      throw new IllegalArgumentException("Join keys can not be empty");
+    }
+
+    Iterable<String> multipleJoinKeys = Splitter.on('&').trimResults().omitEmptyStrings().split(joinKeys);
+
+    if (Iterables.isEmpty(multipleJoinKeys)) {
+      throw new IllegalArgumentException("Join keys can not be empty.");
+    }
+
+    int numJoinKeys = 0;
+    for (String singleJoinKey : multipleJoinKeys) {
+      KeyValueListParser kvParser = new KeyValueListParser("\\s*=\\s*", "\\.");
+      Iterable<KeyValue<String, String>> keyValues = kvParser.parse(singleJoinKey);
+      if (numJoinKeys == 0) {
+        numJoinKeys = Iterables.size(keyValues);
+      } else if (numJoinKeys != Iterables.size(keyValues)) {
+        throw new IllegalArgumentException("There should be one join key from each of the stages. Please add join " +
+                                             "keys for each stage.");
+      }
+      for (KeyValue<String, String> keyValue : keyValues) {
+        String stageName = keyValue.getKey();
+        String joinKey = keyValue.getValue();
+        if (!stageToKey.containsKey(stageName)) {
+          stageToKey.put(stageName, new ArrayList<String>());
+        }
+        stageToKey.get(stageName).add(joinKey);
+      }
+    }
+    return stageToKey;
+  }
+
+  Table<String, String, String> getPerStageSelectedFields() {
+    // table to store <stageName, oldFieldName, alias>
+    ImmutableTable.Builder<String, String, String> tableBuilder = new ImmutableTable.Builder<>();
+
+    if (Strings.isNullOrEmpty(selectedFields)) {
+      throw new IllegalArgumentException("selectedFields can not be empty. Please provide at least 1 selectedFields");
+    }
+
+    for (String selectedField : Splitter.on(',').trimResults().omitEmptyStrings().split(selectedFields)) {
+      Iterable<String> stageOldNameAliasPair = Splitter.on(" as ").trimResults().omitEmptyStrings()
+        .split(selectedField);
+      if (Iterables.size(stageOldNameAliasPair) != 2) {
```

Comment:
we shouldn't require people to use ' as '. Instead, if ' as ' is not used, you can just have the alias be the same as the field name.